app/assets/javascripts/user_analytics.js

```diff
@@ -0,0 +1,182 @@
+//
+// stats
+//
+function statistics() {
+  return JSON.parse(document
+    .getElementById('statistics')
+    .attributes
+    .getNamedItem('data-statistics')
+    .value);
+}
+
+function dailyStatistics() {
+  return Object.entries(statistics()
+    .daily
+    .grouped_by_date);
+}
+
+//
+// dates
+//
+function formatDate(date) {
+  return date.toJSON().slice(0, 10);
+}
+
+function today() {
+  return new Date();
+}
+
+function yesterday() {
+  const todayDate = today();
```

Comment:
Is the ES6 syntax ok? Is it transpiled to ES5? Will apps support this in their webview?

Reply:
As I understand it `> API 21`  supports ES6. I will confirm this with the mobile team.